src/aggsig.rs

```diff
@@ -111,6 +111,10 @@ pub fn verify_single(secp: &Secp256k1, sig:&Signature, msg:&Message, pubnonce:Op
         false => 0,
     };
 
+    if (sig.0).0.ends_with(&[0u8; 32]) || (pubkey.0).0.starts_with(&[0u8; 32]) {
```

Comment:
Do you really mean to only test 32 byte? A signature is 64 bytes and a pubkey 33.

Reply:
Oh also maybe define those zero arrays as const to avoid re-allocating every time.

Reply:
- Yes, only test 32 bytes.
- pubkey is also 64:
```
pub struct PublicKey(pub [c_uchar; 64]);
```
- Because for Signature, actually it contains 2 parts: `(s, r)`,  only `r` is public key related.  0 value `s` is not a problem for this api.
- And similar reason for PublicKey.

Reply:
> define those zero arrays as const to avoid re-allocating every time

OK.